.github/workflows/VercelDeploy.yaml

```
@@ -4,14 +4,74 @@ on:
+        run: npm ci
+
+      - name: Pull Vercel Environment
+        run: npx vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_CLI_TOKEN }}
+
+      - name: Build project
+        run: vercel build --yes --token=${{ secrets.VERCEL_CLI_TOKEN }}
+        timeout-minutes: 10
+
+      - name: Deploy to Vercel
+        id: deploy
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:5:44: Double quote to prevent globbing and word splitting

(shellcheck)